enhanced-stack-trace-queries.md

```diff
@@ -0,0 +1,84 @@
+# Enhanced stack trace queries
+
+## Backround
+
+The SDKs have a built-in `__stack_trace` query that can be used display where the workflow is blocked at the time of the
+query (the last processed workflow task).
+
+The response to a `__stack_trace` query is a single string representing a list of stack individual stack traces
+separated by double newlines.
+
+Example from the TS SDK:
+
+```
+    at someInternalFunction (/path/to/workflows/my-workflow.ts:18:10)
+    at someWorkflowFunction (/path/to/workflows/my-workflow.ts:60:20)
+
+    at someInternalFunction2 (/path/to/workflows/my-workflow.ts:100:30)
+    at someWorkflowFunction2 (/path/to/workflows/my-workflow.ts:123:5)
+```
+
+While this is valuable for developers to be able to understand how their code correlates to event histories, it is quite
+limited since it requires looking up the actual code to get the full picture.
+
+## Proposal
+
+SDKs can optionally implement an `__enhanced_stack_trace` query (name TBD) where instead of returning the stack traces
+as a single string, they return a pre-defined structure, that structure will also include file sources.
+
+The web UI can use this response to show the workflow code and highlight locations where it is blocked.
+
+### Proposed structure
+
+```ts
+/**
+ * A pointer to a location in a file
+ */
+export interface FileLocation {
+  filename: string;
+  line: number;
+  column: number;
+}
+
+/**
+ * Used as the result for the enhanced stack trace query
+ */
+export interface EnhancedStackTrace {
+  /**
+   * Mapping of file name to file contents
+   */
+  sources: Record<string, string>;
+  locations: FileLocation[];
+}
```

Comment:
```suggestion
  sdk: string;  
}
```

`'typescript' | 'go'` etc

Might be helpful

Reply:
Good idea, but I'd do `sdk: { name, version }`